httplog/logger.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2022 The forwarder Authors. All rights reserved.
+// Use of this source code is governed by a MPL
+// license that can be found in the LICENSE file.
+
+package httplog
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/google/martian/v3/log"
+	"github.com/google/martian/v3/messageview"
+	"github.com/saucelabs/forwarder/middleware"
+)
+
+type LoggerMode string
+
+const (
+	URLLogMode     LoggerMode = "url"
+	HeadersLogMode LoggerMode = "headers"
+	BodyLogMode    LoggerMode = "body"
+	ErrOnlyLogMode LoggerMode = "error"
+)
+
+func (m LoggerMode) String() string {
+	return string(m)
+}
+
+// Logger is a modifier that logs requests and responses.
+type Logger struct {
+	log func(line string)
```

Comment:
How about we used the logging function signature `func(format string, args ...interface{})`